app/Filament/Resources/AchievementResource.php

```diff
@@ -208,6 +228,41 @@ public static function form(Form $form): Form
                                 ->disabled(!$user->can('updateField', [$form->model, 'DisplayOrder'])),
                         ]),
                 ])->from('md'),
+
+                Forms\Components\Fieldset::make('Event Association')
+                    ->relationship('eventData')
+                    ->schema([
+                        Forms\Components\Select::make('source_achievement_id')
```

Comment:
This is looking a lot better. One minor suggestion: I think we may want to change this from a `Fieldset` to a `Section`, and give it a `->columns()` value, like:

```php
Forms\Components\Section::make('Event Association')
    ->relationship('eventData')
    ->columns(['xl' => 2, '2xl' => 4])
    ->schema([
```

After this change, the new section blends in with the others on the page as well as View mode:


![Screenshot 2024-10-30 at 8 19 34 PM](https://github.com/user-attachments/assets/12c694f2-9f03-4cc7-bf40-bcff7fe0878f)